src/govuk/components/exit-this-page/exit-this-page.mjs

```diff
@@ -0,0 +1,343 @@
+/* eslint-disable es-x/no-function-prototype-bind -- Polyfill imported */
+
+import { normaliseDataset } from '../../common/normalise-dataset.mjs'
+import { nodeListForEach, mergeConfigs, extractConfigByNamespace } from '../../common.mjs'
+import { I18n } from '../../i18n.mjs'
+import '../../vendor/polyfills/Element/prototype/classList.mjs'
+import '../../vendor/polyfills/Element/prototype/dataset.mjs'
+import '../../vendor/polyfills/Function/prototype/bind.mjs'
+
+/**
+ * @constant
+ * @type {ExitThisPageTranslations}
+ * @see Default value for {@link ExitThisPageConfig.i18n}
+ * @default
+ */
+var EXIT_THIS_PAGE_TRANSLATIONS = {
+  activated: 'Loading.',
+  timedOut: 'Exit this page expired.',
+  pressTwoMoreTimes: 'Shift, press 2 more times to exit.',
+  pressOneMoreTime: 'Shift, press 1 more time to exit.'
+}
+
+/**
+ * JavaScript functionality for the Exit this Page component
```

Comment:
Can we mark everything in this class other than the constructor and init function as private? In all other classes we've deprecated everything and plan to make them private in v5 – we should  avoid these ever being 'public'.

Reply:
So sadly it looks like we can't do this. A note from @colinrotherham on slack:

>We mainly can’t use `@private` in v4 as the `this` keyword (inside a constructor function or methods) can be `window` until it’s been instantiated. The TypeScript compiler isn’t clever enough at this sort of thing until ES2015 classes come along

The chat's continuing but we might not have a suitable workaround at the moment.